tests/test_main_github.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+import os
+import pytest
+from urlchecker.main.github import clone_repo, delete_repo, get_branch
+
+
+@pytest.mark.parametrize('git_path', ["https://github.com/urlstechie/urlchecker-test-repo"])
+def test_clone_and_del_repo(git_path):
```

Comment:
This should probably also use the tmp_path fixture.

Reply:
I tried to take the fast road but I agree temp files are probably better. 
Quick question tho, if use the temp files, do I need to remove the files afterwards? This issue of having to remove some files caused some errors and limited the tests. For example; I couldn't run the tests in parallel because some functions would delete temp test folders needed in other tests :/

Reply:
Nope that's one of the great things about tmp_path! It actually returns a path object that you can do operations on (e.g., clone a repository there or to a subfolder) and it will be cleaned up when the test finishes. Probably the best thing to do is look at other's examples of using it and decide what you like best.